src/frontmatter.md

```diff
@@ -0,0 +1,39 @@
+r[frontmatter]
+# Frontmatter
+
+r[frontmatter.syntax]
+```grammar,lexer
+@root FRONTMATTER ->
+      FRONTMATTER_FENCE HORIZONTAL_WHITESPACE* INFOSTRING? HORIZONTAL_WHITESPACE* LF
+      (FRONTMATTER_LINE LF )*
+      FRONTMATTER_FENCE[^matched-fence] HORIZONTAL_WHITESPACE* LF
```

Comment:
It's a bit unusual to have load-bearing footnotes in the grammar. Would it be possible to define this recursively so that isn't necessary?

Reply:
I worry that being too clever in encoding this could have a negative impact on users understanding this.

Another route is to extend the grammar for inline context notes like how `*?` gets a `(non-greedy)` note and `!` gets a warning icon with `with the exception of`.